src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
-                html = "";
-            html += '<h3 class="sidebarHeader">', html += globalize.translate("HeaderMedia"), html += "</h3>", html += items.map(function(i) {
-                var icon = "folder",
-                    itemId = i.Id;
-                "channels" == i.CollectionType ? itemId = "channels" : "livetv" == i.CollectionType && (itemId = "livetv"), "photos" == i.CollectionType ? icon = "photo_library" : "music" == i.CollectionType || "musicvideos" == i.CollectionType ? icon = "library_music" : "books" == i.CollectionType ? icon = "library_books" : "playlists" == i.CollectionType ? icon = "view_list" : "games" == i.CollectionType ? icon = "games" : "movies" == i.CollectionType ? icon = "video_library" : "channels" == i.CollectionType || "Channel" == i.Type ? icon = "videocam" : "tvshows" == i.CollectionType ? icon = "tv" : "livetv" == i.CollectionType && (icon = "live_tv"), icon = i.icon || icon;
-                i.onclick && i.onclick;
-                return '<a is="emby-linkbutton" data-itemid="' + itemId + '" class="lnkMediaFolder navMenuOption" href="' + getItemHref(i, i.CollectionType) + '"><i class="md-icon navMenuOptionIcon">' + icon + '</i><span class="sectionName navMenuOptionText">' + i.Name + "</span></a>"
-            }).join(""), libraryMenuOptions.innerHTML = html;
-            for (var elem = libraryMenuOptions, sidebarLinks = elem.querySelectorAll(".navMenuOption"), i = 0, length = sidebarLinks.length; i < length; i++) sidebarLinks[i].removeEventListener("click", onSidebarLinkClick), sidebarLinks[i].addEventListener("click", onSidebarLinkClick)
-        })
+        if (!user) {
+            showBySelector(".libraryMenuDownloads", false);
+            showBySelector(".lnkSyncToOtherDevices", false);
+            return void showBySelector(".userMenuOptions", false);
+        }
+
+        if (user.Policy.EnableContentDownloading) {
+            showBySelector(".lnkSyncToOtherDevices", true);
+        } else {
+            showBySelector(".lnkSyncToOtherDevices", false);
+        }
+
+        if (user.Policy.EnableContentDownloading && appHost.supports("sync")) {
+            showBySelector(".libraryMenuDownloads", true);
+        } else {
+            showBySelector(".libraryMenuDownloads", false);
+        }
+
+        var userId = Dashboard.getCurrentUserId();
+        var apiClient = getCurrentApiClient();
+        var libraryMenuOptions = document.querySelector(".libraryMenuOptions");
+
+        if (libraryMenuOptions) {
+            getUserViews(apiClient, userId).then(function (result) {
+                var items = result;
+                var html = "";
+                html += '<h3 class="sidebarHeader">';
+                html += globalize.translate("HeaderMedia");
+                html += "</h3>";
+                html += items.map(function (i) {
+                    var icon = "folder";
+                    var itemId = i.Id;
+
+                    if ("channels" == i.CollectionType) {
```

Comment:
All these double equals should be triple eg. `"channels" === i.CollectionType` since we know it's a string (it's just convention really)